dask/dataframe/core.py

```diff
@@ -1536,34 +1536,127 @@ def quantile(self, q=0.5, axis=0, method='default'):
                 return DataFrame(graph, keyname, meta, quantiles[0].divisions)
 
     @derived_from(pd.DataFrame)
-    def describe(self, split_every=False, percentiles=None, percentiles_method='default'):
-        """Currently, only numeric describe is supported """
-        num = self._get_numeric_data()
-        if self.ndim == 2 and len(num.columns) == 0:
-            raise ValueError("DataFrame contains only non-numeric data.")
-        elif self.ndim == 1 and self.dtype == 'object':
-            raise ValueError("Cannot compute ``describe`` on object dtype.")
-        if percentiles is None:
-            percentiles = [0.25, 0.5, 0.75]
+    def describe(self,
+                 split_every=False,
+                 percentiles=None,
+                 percentiles_method='default',
+                 include=None,
+                 exclude=None):
+
+        from pandas.api.types import is_bool_dtype, is_timedelta64_dtype, \
+            is_numeric_dtype, is_datetime64_any_dtype
+
+        def describe_numeric(data, split_every=False, percentiles=None,
```

Comment:
Slight preference to define these inner functions outside this method. If you want to group all the describe stuff together, you might even define describe as a function elsewhere as well, and have the method just call the function.

Reply:
thank you for your comments @jcrist . I can move them outside and rename to "_<func_name>" as one option.  Would this be ok this way? 
I thought about moving these helper functions outside dataframe/core but didn't find appropriate place for them among existing modules. Maybe you have some suggestions?

Reply:
>  I can move them outside and rename to "_<func_name>" as one option. Would this be ok this way?

That'd be a fine solution.

> I thought about moving these helper functions outside dataframe/core but didn't find appropriate place for them among existing modules. Maybe you have some suggestions?

At this point we don't really have a better place for things like this. For now I'd leave them in `dask/dataframe/core.py`.